### PR TITLE
Use Argparse in mdpow-pcw and mdpow-pow and add loggers

### LINKED ISSUE #147: treatment of use of SI should be transparent
https://github.com/Becksteinlab/MDPOW/pull/142#issuecomment-830683549 indicates that there are still issues in how SI is used and enabled/disabled in scripts (see also #135 )

- It should be easy to enable/disable use of SI.
- The log file needs to contain (at INFO) what settings are used (enable/disabled) and we would also want the actual value.

### PR DESCRIPTION
**Pull request by VOD555**
fix #147 
-replace optparse with argparse
-add loggers to print out settings

### CODE REVIEW

**Comment by orbeckst** [comment]
We don't have argparse in Python 2 https://docs.python.org/3/library/argparse.html so I'll leave it open until we get rid of Python 2.7

**Comment by VOD555** [comment]
@orbeckst It's in Python 2.7 https://docs.python.org/2.7/library/argparse.html

**Comment by orbeckst** [comment]
Oh, I only saw the note in the 3.x page that said “added in 3.3”. That’s good then.

**Comment by orbeckst** [comment]
ping me when you need a review

**Comment by VOD555** [comment]
Current code cannot deal with `--SI` and  `--no-SI` correctly. I'll use Argument groups for them.

**Comment by VOD555** [comment]
@orbeckst This PR can be reviewed.

File: scripts/mdpow-pcw [hunk 1]
```diff
@@ -183,61 +183,71 @@ def realpath_or_none(fn):
...
+    parser = argparse.ArgumentParser(usage=__doc__, prog='mdpow-pcw',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("directory", nargs='+',
+                        help="directory or directories which contain all the files "
+                        "resulting from running mdpow.fep.Ghyd.setup() and mdpow.fep.Gcylo.setup()"
+                        "and the results of the MD FEP simulations.",)
+    parser.add_argument('--plotfile', dest="plotfile", default='none',
+                        help="plot dV/dlambda to FILE; use png or pdf suffix to "
+                        "determine the file type. 'auto' generates a pdf file "
+                        "DIRECTORY/dVdl_MOLID_pcw.pdf and 'None' disables it "
```
Comment (orbeckst) [inline scripts/mdpow-pcw:197]:
```suggestion
                        "DIRECTORY/dVdl_MOLID_pcw.pdf and 'none' disables it "
```

File: scripts/mdpow-pow [hunk 2]
```diff
@@ -183,61 +183,71 @@ def realpath_or_none(fn):
...
+    parser = argparse.ArgumentParser(usage=__doc__, prog='mdpow-pow',
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("directory", nargs='+',
+                        help="directory or directories which contain all the files "
+                        "resulting from running mdpow.fep.Ghyd.setup() and mdpow.fep.Goct.setup()"
+                        "and the results of the MD FEP simulations.",)
+    parser.add_argument('--plotfile', dest="plotfile", default='none',
+                        help="plot dV/dlambda to FILE; use png or pdf suffix to "
+                        "determine the file type. 'auto' generates a pdf file "
+                        "DIRECTORY/dVdl_MOLID_pcw.pdf and 'None' disables it "
```
Comment (orbeckst) [inline scripts/mdpow-pow:197]:
```suggestion
                        "DIRECTORY/dVdl_MOLID_pcw.pdf and 'none' disables it "
```

**Review by orbeckst — APPROVED** [review]
Minor doc corrections to do, which I'll do before merging.

Looking good!

**Comment by orbeckst** [comment]
Note to myself: this is the log message for the squash-merge

* fix #147 
* use exclusive options --SI and --no-SI in the mdpow-pow, mdpow-solvationenergy and mdpow-pcw
* use logging to output use of SI
* replaced optarg with argparse
* fix docs for --plot-file

**Pull request merged**